server/models/room.js

```diff
@@ -38,12 +32,36 @@ const update = roomId => state => {
     }
 }
 
-exports.create = numRounds => {
-    const roomId = latestRoomId;
-    latestRoomId++;
-    const gameState = new GameLogic(update(roomId), roomId, numRounds);
-    rooms.push({ roomId, title: roomNames[Math.floor(Math.random() * 78)], gameState });
-    return roomId;
+exports.create = (numRounds, gameMode) => {
+    return new Promise((resolve, reject) => {
+        const roomId = latestRoomId;
+        latestRoomId++;
+        if (gameMode === 'custom') {
+            db.getAllCards()
+            .then(cards => {                         
+                if (cards.length >= 50) {
+                    const gameState = new GameLogic(update(roomId), roomId, numRounds, gameMode, cards.length);
+                    rooms.push({ roomId, title: roomNames[Math.floor(Math.random() * 78)], gameState });         
+                    resolve(roomId);
+                } else {
+                    reject({
+                        code: 400,
+                        message: "Not enough cards in the server. Upload more."
+                    });
+                }
+            })
+            .catch(err => {
+                reject({
+                    code: err.code,
+                    message: err.message
+                });
+            });
+        } else {
+            const gameState = new GameLogic(update(roomId), roomId, numRounds, gameMode, 247 /*cards length*/);
```

Comment:
comment left in the args here

Reply:
I put it on purpose, thought it would be more clear for someone reading the code, otherwise it's just a number 😄

Reply:
and if the number of original cards change, it'd be easier to find where to change the size of the deck

Reply:
oh sorry I thought it was a separate argument! Is it worth defining that as a const so the variable name speaks for itself?

Reply:
that's a good point, idk why I didn't think of that haha, thanks!

Reply:
If you want clarity, maybe you should make it a variable (constant).